vendors/nuvoton/boards/numaker_iot_m487_wifi/aws_demos/application_code/main_eth.c

```diff
@@ -0,0 +1,417 @@
+   .u.x.ucMinor = APP_VERSION_MINOR,
+   .u.x.usBuild = APP_VERSION_BUILD,
+};
+#endif
+
+/* Sleep on this platform */
+#define Sleep( nMs )    vTaskDelay( pdMS_TO_TICKS( nMs ) );
+
+#define mainDEVICE_NICK_NAME                "Nuvoton_Demo"
+
+#define mainLOGGING_TASK_STACK_SIZE         ( configMINIMAL_STACK_SIZE * 5 )
+#define mainLOGGING_MESSAGE_QUEUE_LENGTH    ( 25 )
+
+
+/* The default IP and MAC address used by the demo.  The address configuration
+ * defined here will be used if ipconfigUSE_DHCP is 0, or if ipconfigUSE_DHCP is
+ * 1 but a DHCP server could not be contacted.  See the online documentation for
+ * more information.  In both cases the node can be discovered using
+ * "ping RTOSDemo". */
+static const uint8_t ucIPAddress[ 4 ] =
+{
+    configIP_ADDR0,
+    configIP_ADDR1,
+    configIP_ADDR2,
+    configIP_ADDR3
+};
+static const uint8_t ucNetMask[ 4 ] =
+{
+    configNET_MASK0,
+    configNET_MASK1,
+    configNET_MASK2,
+    configNET_MASK3
+};
+static const uint8_t ucGatewayAddress[ 4 ] =
+{
+    configGATEWAY_ADDR0,
+    configGATEWAY_ADDR1,
+    configGATEWAY_ADDR2,
+    configGATEWAY_ADDR3
+};
+static const uint8_t ucDNSServerAddress[ 4 ] =
+{
+    configDNS_SERVER_ADDR0,
+    configDNS_SERVER_ADDR1,
+    configDNS_SERVER_ADDR2,
+    configDNS_SERVER_ADDR3
+};
+
+/* Default MAC address configuration.  The demo creates a virtual network
+ * connection that uses this MAC address by accessing the raw Ethernet data
+ * to and from a real network connection on the host PC.  See the
+ * configNETWORK_INTERFACE_TO_USE definition for information on how to configure
+ * the real network connection to use. */
+const uint8_t ucMACAddress[ 6 ] =
+{
+    configMAC_ADDR0,
+    configMAC_ADDR1,
+    configMAC_ADDR2,
+    configMAC_ADDR3,
+    configMAC_ADDR4,
+    configMAC_ADDR5
+};
+
+/**
+ * @brief Application task startup hook.
+ */
+void vApplicationDaemonTaskStartupHook( void );
+
+
+/**
+ * @brief Initializes the board.
+ */
+static void prvMiscInitialization( void );
+
+/*-----------------------------------------------------------*/
+/**
+ * @Set up the hardware ready to run this demo.
+ */
+static void prvSetupHardware( void );
+/*-----------------------------------------------------------*/
+
+
+/**
+ * @brief Application runtime entry point.
+ */
+#define mainCHECK_TASK_PRIORITY             ( tskIDLE_PRIORITY + 3UL )
+
+#define mainCHECK_TASK_STACK_SIZE           ( configMINIMAL_STACK_SIZE )
+#define mainCHECK_DELAY                     ( ( portTickType ) 5000 / portTICK_RATE_MS )
+static void vCheckTask( void *pvParameters )
+{
+    portTickType xLastExecutionTime;
+
+    xLastExecutionTime = xTaskGetTickCount();
+
+    printf("Check Task is running ...\n");
+
+    for( ;; )
+    {
+        /* Perform this check every mainCHECK_DELAY milliseconds. */
+        vTaskDelayUntil( &xLastExecutionTime, mainCHECK_DELAY );
+				printf("Check delay ...%d\n",xTaskGetTickCount());
+			  Sleep( 1 );
+    }
+}
+ 
+int main( void )
```

Comment:
Not sure why you create a new main function, instead of reusing the current one?

Reply:
@qiutongs  It's a little complex to merge wifi & eth in one main file for user view point. So, I split as standalone for each connectivity.

Reply:
hmm, then how would user select which main function to run?

Reply:
@qiutongs  To avoid confusion, I merged these 2 main functions to one main.c in commit c372ef45